### ISSUE TITLE
Be able to access raw HTTP/2 stream

### ISSUE THREAD

**Issue by JensRantil** [golang/go#34874]
I am interested in creating a HTTP proxy (note: _not_ a reverse proxy) in Go. I'm only interested in implementing the `CONNECT` method. All other methods are irrelevant for me. https://github.com/elazarl/goproxy comes very close to what I want to do, but I'm also interested in [supporting `CONNECT` when client is connected to proxy using HTTP/2](https://github.com/elazarl/goproxy/issues/361). AFAIK, according to RFC7540, `CONNECT` over HTTP/2 [should work by hijacking the stream and do TCP tunneling over it](https://httpwg.org/specs/rfc7540.html#CONNECT).

[*http2responseWriter does not implement http.Hijacker](#14797). One idea, would be that it implements `http.Hijacker` which will give access not to the underlying TCP connection but rather the HTTP stream.

...or is there any other approach I could take without having to implement low-level HTTP/2 stuff?

**Comment by agnivade** [comment]
Hello @JensRantil, unlike other projects, we do not use the issue tracker for questions such as these. It is only used for bugs and feature proposals. I will close this issue, but please feel free to ask it in any of these forums below:

* [Stack Overflow](https://stackoverflow.com/questions/tagged/go) with questions tagged "go"

* [The Go Forum](https://forum.golangbridge.org/), a web-based forum

* [Gophers Slack](https://gophers.slack.com), use the [invite app](https://invite.slack.golangbridge.org/) for access. The `#general` channel is a good starting point.

* [Go Community on Hashnode](https://hashnode.com/n/go) with questions and posts tagged with "go"

* [The golang-nuts mailing list](https://groups.google.com/d/forum/golang-nuts)

* **IRC** channel #go-nuts on Freenode

Thanks

**Comment by JensRantil** [comment]
@agnivade Maybe I should have phrased it differently, then. See this as a feature request being able to access a raw stream in HTTP/2.

**Comment by agnivade** [comment]
Ok.

@bradfitz

**Comment by Ice3man543** [comment]
Any updates on this? It'd be very useful to have in order to build proxies that work for HTTP/2 too!

**Comment by Impact-I** [comment]
Any Update?

**Comment by becheran** [comment]
I am also interested in hijacking http2 streams for a tunnel solution based on individual http2 streams. Already created a fork which allows to hijack a http2 stream in the handle method https://github.com/becheran/net. I guess it would make sense to not use the original `http.Hijacker` interface but something stream specific instead to not be confused.

Will this ever be a feature in the http2 lib?

**Comment by edena-legit** [comment]
[JensRantil](https://github.com/JensRantil) anything on that?

**Comment by JensRantil** [comment]
@edena-legit On what? This issue?

**Comment by edena-legit** [comment]
@JensRantil  yes, im trying to use goproxy to proxy have access to http2 streams.. saw that you are too. have you succeeded?

**Comment by JensRantil** [comment]
@edena-legit I have not. I am no longer actively tracking this issue and have little interest in it at the moment.
